crates/core/src/kernel/snapshot/log_segment.rs

```diff
@@ -480,6 +480,7 @@ async fn list_log_files_with_checkpoint(
     let version_prefix = format!("{:020}", cp.version);
     let start_from = log_root.child(version_prefix.as_str());
 
+    // QUESTION: DOES THIS NEED TO BE FILTERED TO ELIMINATE temp subdirs?
```

Comment:
I think this function, plus the one below may need to be fixed as well to ignore temp subdirectories but I could use some feedback here + pointers to the associated unit test.